drivers/flash/soc_flash_nrf5.c

```diff
@@ -161,12 +199,357 @@ static const struct flash_driver_api flash_nrf5_api = {
 	.write_protection = flash_nrf5_write_protection,
 };
 
+
+
 static int nrf5_flash_init(struct device *dev)
 {
 	dev->driver_api = &flash_nrf5_api;
 
+	k_sem_init(&sem_lock, 1, 1);
+
+#if defined(CONFIG_SOC_FLASH_NRF5_RADIO_SYNC)
+	k_sem_init(&sem_sync, 0, 1);
+#endif /* CONFIG_SOC_FLASH_NRF5_RADIO_SYNC */
+
 	return 0;
 }
 
 DEVICE_INIT(nrf5_flash, CONFIG_SOC_FLASH_NRF5_DEV_NAME, nrf5_flash_init,
 	     NULL, NULL, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+
+
+#if defined(CONFIG_SOC_FLASH_NRF5_RADIO_SYNC)
+
+static void time_slot_callback_work(u32_t ticks_at_expire, u32_t remainder,
+		u16_t lazy,
+		void *context)
+{
+	int result;
+	u8_t ticker_id;
+	u8_t instance_index;
+	struct flash_op_desc *op_desc;
+
+	if (ll_radio_state_is_idle()) {
```

Comment:
err = ll_radio_state_is_idle();
__ASSERT(!err);

Then you dont need if-clause, and will reduce indentation.